examples/correlated_ou_1d/analysis_ld.jl

```diff
@@ -0,0 +1,159 @@
+using BSON
+using CUDA
+using Flux
+using Images
+using ProgressMeter
+using Plots
+using Random
+using Statistics
+using TOML
+
+using CliMAgen
+package_dir = pkgdir(CliMAgen)
+include(joinpath(package_dir,"examples/utils_data.jl"))
+include(joinpath(package_dir,"examples/utils_analysis.jl"))
+include(joinpath(package_dir,"examples/correlated_ou_1d/store_load_samples.jl"))
+
+function run_analysis(params; FT=Float32, logger=nothing)
+    # unpack params
+    savedir = params.experiment.savedir
+    rngseed = params.experiment.rngseed
+    nogpu = params.experiment.nogpu
+
+    batchsize = params.data.batchsize
+    fraction = params.data.fraction
+    standard_scaling = params.data.standard_scaling
+    preprocess_params_file = joinpath(savedir, "preprocessing_standard_scaling_$standard_scaling.jld2")
+    n_pixels = params.data.n_pixels
+    n_time = params.data.n_time
+    @assert n_pixels == n_time
+    inchannels = params.model.inchannels
+
+    make_samples = params.sampling.make_samples
+    samples_file = params.sampling.samples_file
+    nsamples = params.sampling.nsamples
+    nimages = params.sampling.nimages
+    nsteps = params.sampling.nsteps
+    sampler = params.sampling.sampler
+    k_bias = params.sampling.k_bias
+
+    # set up rng
+    rngseed > 0 && Random.seed!(rngseed)
+
+    # set up device
+    if !nogpu && CUDA.has_cuda()
+        device = Flux.gpu
+        @info "Sampling on GPU"
+    else
+        device = Flux.cpu
+        @info "Sampling on CPU"
+    end
+
+    # set up dataset
+    dl,_ = get_data_correlated_ou1d(batchsize,preprocess_params_file;
+                                    f=fraction,
+                                    n_pixels=n_pixels,
+                                    n_time=n_time,
+                                    standard_scaling=standard_scaling,
+                                    read =true,
+                                    save=false,
+                                    FT=FT
+                                    )
+    xtrain = cat([x for x in dl]..., dims=4)
+    # To use Images.Gray, we need the input to be between 0 and 1.
+    # Obtain max and min here using the whole data set
+    maxtrain = maximum(xtrain, dims=(1, 2, 4))
+    mintrain = minimum(xtrain, dims=(1, 2, 4))
+    
+    # To compare statistics from samples and training data,
+    # cut training data to length nsamples.
+    xtrain = xtrain[:, :, :, 1:nsamples]
+
+    if make_samples
+        # set up model
+        checkpoint_path = joinpath(savedir, "checkpoint.bson")
+        BSON.@load checkpoint_path model model_smooth opt opt_smooth
+        model = device(model)
+        
+        # set up bias for space-time mean
+        bias = zeros(FT, n_pixels, n_time, inchannels)
```

Comment:
looks good! d(mean(x))dx = 1.

The indexing of `bias` indicates that we want to bias the middle half of the pixels in time and space.